src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
Yea that should work. I'd just noticed that we also need to copy the config object upon in initializing the window selector object. Otherwise we'll modify the config object and people might want to keep reusing the same object for multiple runs.